component/src/org/wso2/ballerina/connectors/oauth2/ClientConnector.bal

```diff
@@ -44,13 +44,13 @@ public connector ClientConnector (string baseUrl, string accessToken, string cli
     @Return { value:"response object"}
     @Return { value:"Error occured during HTTP client invocation." }
     action post (string path, http:Request originalRequest) (http:Response, http:HttpConnectorError) {
-        json originalPayload = originalRequest.getJsonPayload();
+        var originalPayload = originalRequest.getBinaryPayload();
```

Comment:
What's the reason for changing this from JSON to binary?

Reply:
if 'originalRequest' don't have a JSON payload this will throw an error(ie - for empty http requests ). Is this suppose to support only requests JSON payload?